### PR TITLE
[WIP] Eager update

### PR DESCRIPTION
**Pull request by Nemesisesq**

### CODE REVIEW

File: associations/association.go [hunk 1]
```diff
@@ -106,6 +120,18 @@ func (a Associations) AssociationsBeforeCreatable() []AssociationBeforeCreatable
 	return before
 }
 
+// AssociationsAfterUpdateable returns all associations the implement the AssociationBeforeUpdatable
```
Comment (bebbs) [inline associations/association.go]:
```suggestion
// AssociationsBeforeUpdateable returns all associations the implement the AssociationBeforeUpdatable
```

**Comment by mattboll** [comment]
Hi, I've quickly tested and found a bug, with a many to many relation it doesn't delete the last object. For example, delete all members from team and eager save

```
type Team struct {
Members     Users        `many_to_many:"teams_members" json:"members" db:"-"`
}

```
but it works mostly find, thanks for that :)

**Comment by sunho** [comment]
What is the status of this? It would be amazing if pop supports eager update.

**Comment by sunho** [comment]
If it's okay, I am eager to continue the work.

**Comment by stanislas-m** [comment]
@sunho I'm not against that, but please propose a spec here https://github.com/gobuffalo/pop/issues/136 before rushing to the implementation. :)

Eager update is not a trivial thing, so it must have a proper definition.

**Pull request closed without merging**